site/src/pages/HealthPage/DERPPage.tsx

```diff
@@ -0,0 +1,122 @@
+import { Link, useOutletContext } from "react-router-dom";
+import {
+  Header,
+  HeaderTitle,
+  Main,
+  SectionLabel,
+  BooleanPill,
+  Logs,
+  HealthyDot,
+} from "./Content";
+import { HealthcheckReport } from "api/typesGenerated";
+import Button from "@mui/material/Button";
+import LocationOnOutlined from "@mui/icons-material/LocationOnOutlined";
+import useTheme from "@mui/styles/useTheme";
+import { healthyColor } from "./healthyColor";
+import { Alert } from "components/Alert/Alert";
+import { Helmet } from "react-helmet-async";
+import { pageTitle } from "utils/page";
+
+const flags = [
+  "UDP",
+  "IPv6",
+  "IPv4",
+  "IPv6CanSend",
+  "IPv4CanSend",
+  "OSHasIPv6",
+  "ICMPv4",
+  "MappingVariesByDestIP",
+  "HairPinning",
+  "UPnP",
+  "PMP",
+  "PCP",
+];
+
+export const DERPPage = () => {
+  const { derp } = useOutletContext<HealthcheckReport>();
+  const { netcheck, regions, netcheck_logs: logs } = derp;
+  const theme = useTheme();
+
+  return (
+    <>
+      <Helmet>
+        <title>{pageTitle("DERP - Health")}</title>
+      </Helmet>
+
+      <Header>
+        <HeaderTitle>
+          <HealthyDot
+            healthy={derp.healthy}
+            hasWarnings={derp.warnings.length > 0}
+          />
+          DERP
+        </HeaderTitle>
+      </Header>
+
+      <Main>
+        {derp.warnings.map((warning, i) => {
+          return (
+            <Alert key={i} severity="warning">
+              {warning}
+            </Alert>
+          );
+        })}
+
+        <section>
+          <SectionLabel>Flags</SectionLabel>
+          <div css={{ display: "flex", flexWrap: "wrap", gap: 12 }}>
+            {flags.map((flag) => (
+              <BooleanPill key={flag} value={netcheck[flag]}>
```

Comment:
I think there needs to be an `undefined` check here

`netcheck` is optional, but since its defined type is `any`, the full type is `undefined | any`, which collapses down to `any`, and the type warnings get silenced

Reply:
I think it is ok because if it is undefined, we can understand it is `false`

Reply:
I see, but we can rely on `netcheck` always being present in the payload. I think we didn't type it correctly because it is coming from a third-party service, in this case, Tailscale.

Reply:
Ah, that makes sense. Do you know how hard it is to update the generated types?

Reply:
I wouldn't necessarily rely on it 100%, as the API type is

```
Netcheck     *netcheck.Report `json:"netcheck"`
```

which could be `null`. Not sure if that is the same as `undefined` for your purposes though.

Reply:
In this case, it would have the same risk. Trying to access a property from it would throw an error

Reply:
Gotcha. For now, my recommendation would be to check that it's not null.
In future, we might want to make and use a `codersdk.NetcheckReport` for this, but that adds more LoC than a simple null check.

Reply:
Hm.. ok. I will send a hotfix to check if netcheck is null.